typings/index.d.ts

```diff
@@ -431,6 +431,54 @@ export class CommandInteraction extends Interaction {
   private _createOptionsCollection(options: unknown, resolved: unknown): Collection<string, CommandInteractionOption>;
 }
 
+export class CommandInteractionOptionResolver {
+  public constructor(client: Client, options: CommandInteractionOption[]);
+  public readonly client: Client;
+  private _options: CommandInteractionOption[];
+  private _group: string | null;
+  private _subCommand: string | null;
+  private _getTypedOption(
+    name: string,
+    types: ApplicationCommandOptionType[],
+    properties: (keyof ApplicationCommandOption)[],
+    required: true,
+  ): CommandInteractionOption;
+  private _getTypedOption(
+    name: string,
+    types: ApplicationCommandOptionType[],
+    properties: (keyof ApplicationCommandOption)[],
+    required: boolean,
+  ): CommandInteractionOption | null;
+
+  public get(name: string, required: true): CommandInteractionOption;
+  public get(name: string, required?: boolean): CommandInteractionOption | null;
+  public getSubCommand(): string;
+  public getSubCommandGroup(required: true): string;
+  public getSubCommandGroup(required?: boolean): string | null;
+  public getBoolean(name: string, required: true): boolean;
+  public getBoolean(name: string, required?: boolean): boolean | null;
+  public getChannel(name: string, required: true): NonNullable<CommandInteractionOption['channel']>;
+  public getChannel(name: string, required?: boolean): NonNullable<CommandInteractionOption['channel']> | null;
+  public getString(name: string, required: true): string;
+  public getString(name: string, required?: boolean): string | null;
+  public getInteger(name: string, required: true): number;
+  public getInteger(name: string, required?: boolean): number | null;
+  public getUser(name: string, required: true): NonNullable<CommandInteractionOption['user']>;
+  public getUser(name: string, required?: boolean): NonNullable<CommandInteractionOption['user']> | null;
+  public getMember(name: string, required: true): NonNullable<CommandInteractionOption['member']>;
+  public getMember(name: string, required?: boolean): NonNullable<CommandInteractionOption['member']> | null;
+  public getRole(name: string, required: true): NonNullable<CommandInteractionOption['role']>;
+  public getRole(name: string, required?: boolean): NonNullable<CommandInteractionOption['role']> | null;
+  public getMentionable(
+    name: string,
+    required: true,
+  ): NonNullable<CommandInteractionOption['member' | 'role' | 'user']>;
+  public getMentionable(
+    name: string,
+    required?: boolean,
+  ): NonNullable<CommandInteractionOption['member' | 'role' | 'user']> | null;
```

Comment:
```suggestion
  public getBoolean(name: string, required?: boolean): boolean | undefined;
  public getChannel(name: string, required: true): NonNullable<CommandInteractionOption['channel']>;
  public getChannel(name: string, required?: boolean): NonNullable<CommandInteractionOption['channel']> | undefined;
  public getString(name: string, required: true): string;
  public getString(name: string, required?: boolean): string | undefined;
  public getInteger(name: string, required: true): number;
  public getInteger(name: string, required?: boolean): number | undefined;
  public getUser(name: string, required: true): NonNullable<CommandInteractionOption['user']>;
  public getUser(name: string, required?: boolean): NonNullable<CommandInteractionOption['user']> | undefined;
  public getMember(name: string, required: true): NonNullable<CommandInteractionOption['member']>;
  public getMember(name: string, required?: boolean): NonNullable<CommandInteractionOption['member']> | undefined;
  public getRole(name: string, required: true): NonNullable<CommandInteractionOption['role']>;
  public getRole(name: string, required?: boolean): NonNullable<CommandInteractionOption['role']> | undefined;
  public getMentionable(
    name: string,
    required: true,
  ): NonNullable<CommandInteractionOption['member' | 'role' | 'user']>;
  public getMentionable(
    name: string,
    required?: boolean,
  ): NonNullable<CommandInteractionOption['member' | 'role' | 'user']> | undefined;
```

Reply:
remove `NonNullable` instead of `| undefined` - because `NonNullable` was originally used here to exclude undefined

Reply:
done